serverless/src/getKeywordsTree/handler.js

```diff
@@ -0,0 +1,265 @@
+import { format } from 'date-fns'
+
+import { buildKeywordsTree } from '@/shared/buildKeywordsTree'
+import { filterKeywordTree } from '@/shared/filterKeywordTree'
+import { filterScienceKeywordsTree } from '@/shared/filterScienceKeywordsTree'
+import { getConceptSchemeDetails } from '@/shared/getConceptSchemeDetails'
+import { getApplicationConfig } from '@/shared/getConfig'
+import { getNarrowersMap } from '@/shared/getNarrowersMap'
+import { getRootConceptForScheme } from '@/shared/getRootConceptForScheme'
+import { getRootConceptsForAllSchemes } from '@/shared/getRootConceptsForAllSchemes'
+import { sortKeywordNodes } from '@/shared/sortKeywordNodes'
+import { keywordSchemeSequence, sortKeywordSchemes } from '@/shared/sortKeywordSchemes'
+import { toTitleCase } from '@/shared/toTitleCase'
+
+/**
+ * Retrieves and processes a keywords tree based on the provided concept scheme.
+ *
+ * @param {Object} event - The event object containing query and path parameters.
+ * @param {Object} event.queryStringParameters - Query string parameters.
+ * @param {string} [event.queryStringParameters.filter] - Optional filter string to apply to the tree.
+ * @param {Object} event.pathParameters - Path parameters.
+ * @param {string} event.pathParameters.conceptScheme - The concept scheme to retrieve keywords for.
+ *
+ * @returns {Object} An object containing the status code, body, and headers.
+ *
+ * @example
+ * // Request for all schemes
+ * const event = {
+ *   pathParameters: { conceptScheme: 'all' },
+ *   queryStringParameters: { filter: 'atmosphere' }
+ * };
+ * const result = await getKeywordsTree(event);
+ * // Result will contain a tree of all keyword schemes, filtered by 'atmosphere'
+ *
+ * @example
+ * // Request for Earth Science scheme
+ * const event = {
+ *   pathParameters: { conceptScheme: 'earth science' },
```

Comment:
there is no concept scheme "earth science"?

Reply:
we have scheme 'sciencekeywords' and the 2 direct descendants are 'EARTH SCIENCE' and 'EARTH SCIENCE SERVICES'. We create the schemes 'Earth Science' and 'Earth Science Services' in Keyword Viewer only to distinguish the branches.